test/functional/JIT_Test/src/jit/test/jar/ZipTester.java

```diff
@@ -53,23 +58,31 @@
 	public static void main(String args[]) {
 		new ZipTester().run(args);
 	}
-	
+
 	public void run(String args[]) {
 		if (args.length < 1 || args[0].isEmpty()) {
 			printUsageText();
 			Assert.fail();
 		}
 
-		String classFilter = args[0];
+		String classFilter = "";
+		boolean loopForever = false;
 		int zipFilenameIndex = 0;
-		if (!classFilter.startsWith(CLASS_FILTER_TAG))
-			classFilter = "";
-		else {
-			classFilter = classFilter.substring(CLASS_FILTER_TAG.length());
-			zipFilenameIndex = 1;
+
+		while (args[zipFilenameIndex].startsWith(TAG_PREFIX)) {
+			if (args[zipFilenameIndex].startsWith(CLASS_FILTER_TAG))
+				classFilter = classFilter.substring(CLASS_FILTER_TAG.length());
+			else if (args[zipFilenameIndex].equals(LOOP_FOREVER_TAG))
+				loopForever = true;
+			zipFilenameIndex++;
 		}
 
-		process(classFilter, args, zipFilenameIndex);
+		if (loopForever)
+			logger.info(LOOP_FOREVER_TAG + " was specified; test will run continuously until the JVM is terminated.");
+
+		do {
+			process(classFilter, args, zipFilenameIndex);
```

Comment:
If the process runs forever, who shuts it down? Timeout?

Reply:
The harness which starts/stops the clients/servers sends a `SIGTERM` to stop them. The timing is driven by the harness not the client; the harness assumes the client is running some repetitive task that keeps generating compilation activity and runs clients and servers for various time periods, i.e. "think time".

By looping forever we just ensure that compilation activity is happening for the entire duration of the think time rather than the client finishing really quickly on a fast machine and leaving an idle server sitting around for the rest of the test time period.